scripts/cms5/phpstan.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+// Only valid for non-theme modules
+if (!is_module() || is_theme()) {
+    return;
+}
+
+// Get the dirs we want to run static analysis against
+$dirs = [];
+foreach (['code', 'src', 'app/src'] as $codeDir) {
+    if (check_dir_exists($codeDir)) {
+        $dirs[] = $codeDir;
+    }
+}
+
+// If we have some dirs to run against, we don't need to do anything.
+if (empty($dirs)) {
+    return;
+}
+
+$content = <<<EOT
+parameters:
+  paths:
+    - %s
+
+EOT;
+
+// Create a phpstan config file
+write_file_even_if_exists('phpstan.neon.dist', sprintf($content, implode("\n    - ", $dirs)));
+
+// Add composer dependencies.
+// composer.json file is already guaranteed to exist - it's used in the is_module()/is_theme() checks above.
+// Do not add allow-plugins config - we don't want to be enforcing that for peoples' projects
+$contents = read_file('composer.json');
+$json = json_decode($contents, true);
+$jsonOrig = $json;
+if (!$json) {
+    warning('Failed to parse composer.json');
```

Comment:
Will this hard fail? I think a hard fail is probably better at this point

Reply:
I just copied this logic from the `composer.php` script. 
There is an `error` function, but that's in `funcs_utils.php` which
> THESE FUNCTIONS ARE NOT INTENDED FOR USE IN SCRIPT FILES - only use functions in funcs_scripts.php
> error() is not intended for use in scripts specificially because it will halt progress
Instead scripts should be more tolerant though use of functions like check_file_exists() or just using warning() instead which output a big warning in the console while still continuing